### ISSUE TITLE
[BUG]:ValueError: 'fill_value'=constant is invalid. Expected a numerical value when imputing numerical data

### ISSUE THREAD

**Issue by Prashavu** [pycaret/pycaret#2872]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

This error is thrown in Setup when Preprocess = True and fix_imbalance_method=smote and fix_imbalance=True for classification

### Reproducible Example

```python
No `object` columns are present in the dataframe. All are int/float
```

### Expected Behavior

No Error

### Actual Results

```python-traceback
---> 79 fix_imbalance_method=smote,fix_imbalance=True)

10 frames
/usr/local/lib/python3.7/dist-packages/sklearn/impute/_base.py in fit(self, X, y)
    338                 "'fill_value'={0} is invalid. Expected a "
    339                 "numerical value when imputing numerical "
--> 340                 "data".format(fill_value)
    341             )
    342 

ValueError: 'fill_value'=constant is invalid. Expected a numerical value when imputing numerical data
```

### Installed Versions

<details>
3.0.0rc
</details>

**Comment by ngupta23** [comment]
It is likely now a bug. If you want to fill with a constant value, just provide that value.

**Comment by Prashavu** [comment]
hi @ngupta23 could you elaborate. I am not intending to fill a constant. I think smote is causing this. what package should I update to resolve this issue

**Comment by Prashavu** [comment]
@ngupta23 There are no nulls or NaNs and I am not doing any imputation

**Comment by ngupta23** [comment]
Can you provide a complete reproducible example for this with data.

**Comment by Prashavu** [comment]
Thing is the data is confidential. I will try to reproduce it on the juice data i/f I can. but its not cused by fix_imabalance paramter as even without it , its throwing error

**Comment by Prashavu** [comment]
I think these are the paramters causing this issue. these are some inbuilt conditions handled. For example I do not want any imputation in my data, but maybe another transformation is causing this?
![image](https://user-images.githubusercontent.com/13661149/185380624-dba4e731-8a07-4758-9c4f-bbb0e9b6aadc.png)

**Comment by ngupta23** [comment]
Thanks for digging into it. Which RC version have you installed?

**Comment by Prashavu** [comment]
3.0.0rc

**Comment by ngupta23** [comment]
Can you try to see if it is still there with the latest rc version 3.0.0rc3?

`pip install --pre -U pycaret`

**Comment by Prashavu** [comment]
Yes whatever I'm reporting is with the exact same version

**Comment by tvdboom** [comment]
First, this is probably caused by the imputation step. If you don't want any imputation being done, use `imputation_type=None` in setup. This will probably avoid the error. I don't think `fix_imbalance` has anything to do with it.

Secondly, I can't know for sure if this is really a bug without a reproducible example. It seems you are forcing a categorical feature onto a numerical column. This is forcing pycaret to use an imputer for categorical columns on a feature with an unexpected dtype, possibly creating this error.

**Comment by Prashavu** [comment]
I do not have missing values to perform imputation. So i dint think this is
the reason.

**Comment by Prashavu** [comment]
How do I set these to None further?
![image](https://user-images.githubusercontent.com/13661149/185728921-44f9d192-d9c1-4793-9255-5598add7e8d6.png)

**Comment by tvdboom** [comment]
> I do not have missing values to perform imputation. So i do not  think this is
the reason.I saw in the documentation there is default value for imputation.My data set is 100 p clean, then why is pycaret performing imputation?This should be fixed.

Pycaret has an imputer in the pipeline as default value. Even though you don't have missing values, it could be desired to have an imputer because you could have missing values in new unseen data.  Pycaret is not performing imputation, it is fitting the imputer, which is a different thing. It's not a bug.

To not have any imputer in the pipeline use `imputation_type=None`. The fact that it still shows in the setup output is a bug. I fixed it in #2866. You can check that there are no imputers in the pipeline printing the `pipeline` attribute.  Again, having a reproducible example would make things a lot easier.
